### DISCUSSION TITLE
How to implement Triplanar mapping?

### DISCUSSION THREAD

**Discussion by chaoticgood1** [bevyengine/bevy#818] [Q&A]
Need to implement this on 3D Terrain custom mesh, thanks.

**Comment by bonsairobo** [comment]
I'll let you know once I've implemented it myself ;)

I've done it before in Amethyst, and I'll eventually attempt it in Bevy.

**Comment by chaoticgood1** [comment]
First off, thank you so much for creating voxel-mapper. 

We are using the triplanar code to convert to bevy, so far we are currently solving the issue in the fragment shader. I just realized that you should be the goto guy for this kind of question, hahah!

**Reply by bonsairobo** [reply]
Yea! We're figuring stuff out in the discord. As expected, porting the shaders to bevy is not entirely straightforward.

**Comment by bonsairobo** [comment]
Hey I got this working a while back and forgot to post here. I have both triplanar mapping and biplanar mapping working here:

https://github.com/bonsairobo/smooth-voxel-renderer
